node/network/approval-distribution/src/lib.rs

```diff
@@ -231,6 +376,25 @@ impl Knowledge {
 					true
 				},
 			},
+		};
+
+		// In case of succesful insertion of multiple candidate assignments create additional
+		// entries for each assigned candidate. This fakes knowledge of individual assignments, but
+		// we need to share the same `MessageSubject` with the followup approval candidate index.
+		if kind == MessageKind::Assignment && success && message.1.count_ones() > 1 {
+			message
+				.1
+				.iter_ones()
+				.map(|candidate_index| candidate_index as CandidateIndex)
+				.fold(success, |success, candidate_index| {
+					success &
```

Comment:
I'm confused, why use a bitand `&` here? Is that supposed to be `&&` as a short circuit?
In general, it would be more readable as a for loop I think.

Reply:
made it a for loop